src/crimson/net/Socket.h

```diff
@@ -79,12 +86,25 @@ class Socket
     return out.write(std::move(buf)).then([this] { return out.flush(); });
   }
 
+  // preemptively disable further reads or writes, can only be shutdown once.
+  void shutdown() {
```

Comment:
could you move the implementations into the .cc file?

Reply:
done.